taichi/ir/type.cpp

```diff
@@ -75,4 +75,12 @@ int Type::vector_width() const {
   }
 }
 
+bool Type::is_primitive(PrimitiveTypeID type) const {
+  if (auto p = cast<PrimitiveType>()) {
+    return p->type == type;
+  } else {
+    return false;
+  }
+}
+
```

Comment:
core change.

Besides this function, everything else is just a simple replacement. Though I've checked every replacement, I would like the reviewer to do it again and thanks in advance!